lib/api/bucketUpdateQuota.js

```diff
@@ -0,0 +1,79 @@
+const { waterfall } = require('async');
+const { errors } = require('arsenal');
+const collectCorsHeaders = require('../utilities/collectCorsHeaders');
+const { standardMetadataValidateBucket } = require('../metadata/metadataUtils');
+const metadata = require('../metadata/wrapper');
+const { pushMetric } = require('../utapi/utilities');
+const monitoring = require('../utilities/monitoringHandler');
+
+/**
+ * Bucket Update Quota - Update bucket quota
+ * @param {AuthInfo} authInfo - Instance of AuthInfo class with requester's info
+ * @param {object} request - http request object
+ * @param {object} log - Werelogs logger
+ * @param {function} callback - callback to server
+ * @return {undefined}
+ */
+function bucketUpdateQuota(authInfo, request, log, callback) {
+    log.debug('processing request', { method: 'bucketUpdateQuota' });
+
+    const { bucketName } = request;
+    const metadataValParams = {
+        authInfo,
+        bucketName,
+        requestType: request.apiMethods || 'bucketUpdateQuota',
+        request,
+    };
+    let bucket = null;
+    return waterfall([
+        next => standardMetadataValidateBucket(metadataValParams, request.actionImplicitDenies, log,
+            (err, b) => {
+                bucket = b;
+                return next(err);
+            }),
+        next => {
+            let requestBody;
+            try {
+                requestBody = JSON.parse(request.post);
+            } catch (parseError) {
+                return next(errors.InvalidArgument.customizeDescription('Invalid JSON format in request'));
+            }
+            if (typeof requestBody !== 'object' || Array.isArray(requestBody)) {
+                return next(errors.InvalidArgument.customizeDescription('Request body must be a JSON object'));
+            }
+            const quota = parseInt(requestBody.quota, 10);
+            if (isNaN(quota)) {
+                return next(errors.InvalidArgument.customizeDescription('Quota Value should be a number'));
```

Comment:
These checks could be in a dedicated function, let's say `validateBucketQuotaConfiguration` or something like that; this would also ease the unit testing

Reply:
done here : https://github.com/scality/cloudserver/pull/5552/commits/eeeae66ade78935c691dc25b6a0717de25ffe2f3#diff-6a9079dc9bc522cd0ae740e5325eea15d69b0b818304c3f357a7795cf2dfe63dR22